scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RewriteTrailingCommas.scala

```diff
@@ -0,0 +1,86 @@
+package org.scalafmt.rewrite
+
+import scala.meta._
+import scala.meta.classifiers.Classifier
+import scala.meta.tokens.Token
+
+import org.scalafmt.util.Whitespace
+
+case object RewriteTrailingCommas extends Rewrite {
+
+  override def rewrite(implicit ctx: RewriteCtx): Unit =
+    ctx.tree.traverse {
+      case t: Importer =>
+        t.tokens.lastOption match {
+          case Some(close: Token.RightBrace) =>
+            removeBefore(close)
+          case _ =>
+        }
+
+      case t: Decl.Def => afterTParams(t.tparams); afterEachArgs(t.paramss)
+      case t: Defn.Def => afterTParams(t.tparams); afterEachArgs(t.paramss)
+      case t: Defn.Macro => afterTParams(t.tparams); afterEachArgs(t.paramss)
+      case t: Defn.Class => afterTParams(t.tparams)
+      case t: Defn.Trait => afterTParams(t.tparams)
+      case t: Ctor.Secondary => afterEachArgs(t.paramss)
+      case t: Decl.Type => afterTParams(t.tparams)
+      case t: Defn.Type => afterTParams(t.tparams)
+      case t: Type.Apply => afterArgs(t.args)
+      case t: Type.Param => afterTParams(t.tparams)
+      case t: Type.Tuple => afterArgs(t.args)
+      case t: Term.Function => afterArgs(t.params)
+      case t: Type.Function => afterArgs(t.params)
+      case t: Ctor.Primary => afterEachArgs(t.paramss)
+
+      case t: Term.Apply => afterArgs(t.args)
+      case t: Type.Apply => afterArgs(t.args)
+      case t: Pat.Extract => afterArgs(t.args)
+      case t: Pat.Tuple => afterArgs(t.args)
+      case t: Term.Tuple => afterArgs(t.args)
+      case t: Term.ApplyType => afterArgs(t.targs)
+      case t: Init => afterEachArgs(t.argss)
```

Comment:
I'm not 100% sure that these patterns exhaustively cover all patterns, but looking at the test cases, it's ok I guess :+1:

Reply:
the original code (in `FormatWriter`) checked for `isDefnOrCallSite`, so I took all patterns from `isDefnSite` and `isCallSite` :)